jumpcloud_bootstrap_template.sh

```diff
@@ -987,39 +1149,283 @@ if [[ ! -f $DEP_N_GATE_DONE ]]; then
     echo "Status: System Account Configured" >>"$DEP_N_LOG"
 
     echo "$(date "+%Y-%m-%d %H:%M:%S"): JumpCloud agent local account takeover complete" >>"$DEP_N_DEBUG"
+    echo "Status: Applying Finishing Touches" >>"$DEP_N_LOG"
 
-    # Remove from DEP_ENROLLMENT_GROUP and add to DEP_POST_ENROLLMENT_GROUP
-    removeGrp=$(
-    curl \
-        -X 'POST' \
-        -A "${USER_AGENT}" \
-        -H 'Content-Type: application/json' \
-        -H 'Accept: application/json' \
-        -H 'x-api-key: '${APIKEY}'' \
-        -d '{"op": "remove","type": "system","id": "'${systemID}'"}' \
-        "https://console.jumpcloud.com/api/v2/systemgroups/${DEP_ENROLLMENT_GROUP_ID}/members"
-    )
-    echo "$(date "+%Y-%m-%d %H:%M:%S"): Removing Enrollment Group $removeGrp" >>"$DEP_N_DEBUG"
-    echo "$(date "+%Y-%m-%d %H:%M:%S"): Removed from DEP_ENROLLMENT_GROUP_ID: $DEP_ENROLLMENT_GROUP_ID" >>"$DEP_N_DEBUG"
-    addGrp=$(
-    curl \
-        -X 'POST' \
-        -A "${USER_AGENT}" \
-        -H 'Content-Type: application/json' \
-        -H 'Accept: application/json' \
-        -H 'x-api-key: '${APIKEY}'' \
-        -d '{"op": "add","type": "system","id": "'${systemID}'"}' \
-        "https://console.jumpcloud.com/api/v2/systemgroups/${DEP_POST_ENROLLMENT_GROUP_ID}/members"
+    FINISH_TEXT='This window will automatically close when onboarding is complete.\n\nYou will be taken directly to the log in screen. \n\n Log in with your new account! \n\n'
 
-    )
-    echo "$(date "+%Y-%m-%d %H:%M:%S"): Adding to Post Enrollment Group $addGrp" >>"$DEP_N_DEBUG"
-    echo "$(date "+%Y-%m-%d %H:%M:%S"): Added to from DEP_POST_ENROLLMENT_GROUP_ID: $DEP_POST_ENROLLMENT_GROUP_ID" >>"$DEP_N_DEBUG"
+    echo "Command: MainText: $FINISH_TEXT" >>"$DEP_N_LOG"
+    # wait just a moment
+    # here we want to just note that the newly matched user does not have secure token
+    # validate that the expected user has secure token
+    stStatus=$(sysadminctl -secureTokenStatus "$enrollmentUser" 2>&1)
+    echo $stStatus
+    regex='Secure token is (ENABLED|DISABLED)'
+    if [[ $stStatus =~ $regex ]]; then
+        stStatusResult="${BASH_REMATCH[1]}"
+    fi
+    if [[ $stStatusResult == 'DISABLED' ]]; then
+        echo "$(date "+%Y-%m-%d %H:%M:%S"): $enrollmentUser Secure Token Status: $stStatusResult" >>"$DEP_N_DEBUG"
+        echo "$(date "+%Y-%m-%d %H:%M:%S"): $enrollmentUser must login before being granted Secure Token" >>"$DEP_N_DEBUG"
+    else
+        echo "$(date "+%Y-%m-%d %H:%M:%S"): $enrollmentUser Secure Token Status: $stStatusResult" >>"$DEP_N_DEBUG"
+    fi
 
-    echo "Status: Applying Finishing Touches" >>"$DEP_N_LOG"
+    sleep 10
+    # steps:
+    # hide enrollment users:
+    echo "$(date "+%Y-%m-%d %H:%M:%S"): Hiding Enrollment Users from Login Window" >>"$DEP_N_DEBUG"
+    dscl . create /Users/$ENROLLMENT_USER IsHidden 1
```

Comment:
Here we are hiding these users from the login window, this is a change in previous versions of the tool since they would have been removed from the agent